### PR TITLE
Feature adapt resource collector and make pod informer just watch the right status

### PR DESCRIPTION
**Pull request by nkwangjun**
**What type of PR is this?**
/kind feature

### CODE REVIEW

**Comment by nkwangjun** [comment]
@cathyhongzhang @CoderKevinZhang  This pr will help gs scheduler to adapte the resource collecotr， and get the cache snapshot from the resource collector. And gs scheduler is already tested with the api server.

File: globalscheduler/pkg/scheduler/framework/interfaces/framework.go [hunk 1]
```diff
@@ -391,12 +391,12 @@ func (f *framework) RunScorePlugins(ctx context.Context, state *CycleState, stac
 }
 
 func (f *framework) runScorePlugin(ctx context.Context, pl ScorePlugin, state *CycleState, stack *types.Stack,
-	siteID string) (int64, *Status) {
+	siteCacheInfo *schedulersitecacheinfo.SiteCacheInfo) (int64, *Status) {
 	if !state.ShouldRecordPluginMetrics() {
-		return pl.Score(ctx, state, stack, siteID)
+		return pl.Score(ctx, state, stack, siteCacheInfo)
```
Comment (kimeunju108) [inline globalscheduler/pkg/scheduler/framework/interfaces/framework.go:396]: Is there any difference between line 396 and line 399 ?

File: globalscheduler/pkg/scheduler/framework/interfaces/framework.go [hunk 2]
```diff
@@ -451,12 +451,12 @@ func (f *framework) RunBindPlugins(ctx context.Context, state *CycleState, stack
 }
 
 func (f *framework) runBindPlugin(ctx context.Context, bp BindPlugin, state *CycleState, stack *types.Stack,
-	siteID string) *Status {
+	siteCacheInfo *schedulersitecacheinfo.SiteCacheInfo) *Status {
 	if !state.ShouldRecordPluginMetrics() {
```
Comment (kimeunju108) [inline globalscheduler/pkg/scheduler/framework/interfaces/framework.go:455]: Is this if !state.ShouldRecordPluginMetrics() statement necessary?  
Regardless this if statement, "bp.Bind(ctx, state, stack, siteCacheInfo)" in line 459 will be performed.

File: globalscheduler/pkg/scheduler/framework/interfaces/framework.go (hunk 1 above)
Comment (nkwangjun) [reply globalscheduler/pkg/scheduler/framework/interfaces/framework.go:396]: Yes，you are right. Here I will remove the if !state.ShouldRecordPluginMetrics()

File: globalscheduler/pkg/scheduler/framework/interfaces/framework.go (hunk 2 above)
Comment (nkwangjun) [reply globalscheduler/pkg/scheduler/framework/interfaces/framework.go:455]: Yes，you are right. Here I will remove the if !state.ShouldRecordPluginMetrics()

**Comment by nkwangjun** [comment]
@kimeunju108  it is fixed.

File: globalscheduler/pkg/scheduler/framework/interfaces/framework.go (hunk 2 above)
Comment (nkwangjun) [reply globalscheduler/pkg/scheduler/framework/interfaces/framework.go]: It is fixed. Thanks

File: globalscheduler/pkg/scheduler/framework/interfaces/framework.go (hunk 1 above)
Comment (nkwangjun) [reply globalscheduler/pkg/scheduler/framework/interfaces/framework.go]: It is fixed. Thanks

**Review by kimeunju108 — APPROVED** [review]
LGTM

**Review by cathyhongzhang — APPROVED** [review]

**Pull request merged**